src/main/java/com/google/sps/utility/FreeTimeUtility.java

```diff
@@ -14,49 +14,67 @@
 
 package com.google.sps.utility;
 
-import com.google.sps.data.CalendarDataResponse;
+import com.google.sps.data.CalendarSummaryResponse;
+import com.google.sps.data.TimeHourMin;
+import java.time.Duration;
 import java.util.*;
 import java.util.concurrent.TimeUnit;
 
 /** Class containing the user's free hours. */
 public final class FreeTimeUtility {
 
   private final Date startDate;
-  private List<Long> workHoursPerDay;
-  private List<Long> personalHoursPerDay;
-  private List<DatePair> morningFreeInterval;
-  private List<DatePair> workFreeInterval;
-  private List<DatePair> eveningFreeInterval;
-  private static final long HOUR_MILLI = TimeUnit.HOURS.toMillis(1);
-  private static final long DAY_MILLI = TimeUnit.DAYS.toMillis(1);
-  private static final long PERSONAL_BEGIN = 7 * HOUR_MILLI;
-  private static final long WORK_BEGIN = 10 * HOUR_MILLI;
-  private static final long WORK_END = 18 * HOUR_MILLI;
-  private static final long PERSONAL_END = 23 * HOUR_MILLI;
-  private static final int OFFSET = 7;
+  /**
+   * numDays represents the number of days from the current one (and including it) we are
+   * considering for our calculations It is used to initialize the free interval lists and the size
+   * of the free time lists that are returned
+   */
+  private final int numDays;
+  /**
+   * The lists of time intervals from now (startDate) to the end day. They are divided by morning,
+   * work, and evening periods to allow easy computation of free hours, and later to create events
+   * during work/morning/evening hours
+   */
+  private List<DateInterval> morningFreeInterval;
+
+  private List<DateInterval> workFreeInterval;
+  private List<DateInterval> eveningFreeInterval;
+  private static final int DAYS_IN_WEEK = 7;
 
   /**
-   * Initialize the class with the start day. The work hours are harrrd-coded between 10 AM and 6
-   * PM. The rest of the free hours are between 7 AM and 11 PM.
+   * Initialize the class with the start day. The work hours and personal hours are provided as
+   * parameters of the constructor.
    *
    * @param startTime parameter that gives the time of the start/now.
+   * @param personalBeginHour parameter that gives the hour to begin the personal time
+   * @param workBeginHour parameter that gives the hour to begin the work time
+   * @param workEndHour parameter that gives the hour to begin the work time
    */
-  public FreeTimeUtility(Date startDate) {
+  public FreeTimeUtility(
```

Comment:
I would consider storing the integers in this utility as `public static final` constants, since they are used in other files as well. That way, `personalBeginHour`, `workBeginHour`, etc. do not need to be defined in each servlet

Reply:
The integers you are mentioning are parameters to the constructor, but not global variables of the class. The reason they are parameters is so that when scaling the project, the user can have the option to set these values from the front-end, and so the values would be served from the request.

Reply:
Ah okay yes that makes sense - they won't be hardcoded forever. 

Thanks!